### PR TITLE
Aka task query new

### PR DESCRIPTION
**Pull request by Alihan26**
# Pull Requests Checklist

Assignee:

- \[x\] code commented?
- \[x\] tests added / passed?
- \[x\] are tests covering all cases?
- \[x\] documentation added / updated?
- \[x\] pre-commit checks passed?
- \[x\] pipelines passed?

Reviewer:

- \[x\] correct branch?
- \[x\] are there any code smells?
- \[x\] did pipeline pass?
- \[x\] is the code well documented?
- \[x\] are the tests passing and covering all cases?

### CODE REVIEW

**Comment by Alihan26** [comment]
@5v3nn the URI ending has been checked with an additional test case and a small modification into the team controller. Ready to merge by your approval.

File: src/test/java/ch/uzh/ifi/hase/soprafs24/controller/TeamControllerTest.java [hunk 1]
```diff
@@ -698,6 +699,106 @@ public void getTasks_unauthorizedAccess_throwsError() throws Exception {
...
+  public void getTasksByStatus_invalidStatus_throwsError() throws Exception {
+    // given
+    Mockito
+        .when(authorizationService.isAuthorizedAndBelongsToTeam(
+            Mockito.anyString(), Mockito.anyLong()))
+        .thenReturn(testUser);
+    Mockito.doThrow(new ResponseStatusException(HttpStatus.CONFLICT, "Invalid task status."))
+        .when(taskService)
+        .getTasksByTeamIdAndStatus(Mockito.anyLong(), Mockito.anyList());
+
```
Comment (5v3nn) [inline src/test/java/ch/uzh/ifi/hase/soprafs24/controller/TeamControllerTest.java:787]: wrong as discussed

File: src/test/java/ch/uzh/ifi/hase/soprafs24/controller/TeamControllerTest.java [hunk 2]
```diff
@@ -698,6 +699,106 @@ public void getTasks_unauthorizedAccess_throwsError() throws Exception {
...
+  @Test
+  public void getTasksByStatus_invalidStatus_throwsError() throws Exception {
+    // given
+    Mockito
+        .when(authorizationService.isAuthorizedAndBelongsToTeam(
+            Mockito.anyString(), Mockito.anyLong()))
+        .thenReturn(testUser);
+    Mockito.doThrow(new ResponseStatusException(HttpStatus.CONFLICT, "Invalid task status."))
+        .when(taskService)
+        .getTasksByTeamIdAndStatus(Mockito.anyLong(), Mockito.anyList());
```
Comment (5v3nn) [inline src/test/java/ch/uzh/ifi/hase/soprafs24/controller/TeamControllerTest.java:786]: wrong as discussed

File: src/test/java/ch/uzh/ifi/hase/soprafs24/controller/TeamControllerTest.java [hunk 3]
```diff
@@ -698,6 +699,106 @@ public void getTasks_unauthorizedAccess_throwsError() throws Exception {
...
+        .willReturn(tasks);
+
+    // when/then -> do the request + validate the result
+    MockHttpServletRequestBuilder getRequest =
+        get("/api/v1/teams/1/tasks?status=TODO").header("Authorization", "1234");
+
+    // then
+    mockMvc.perform(getRequest)
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$[0].taskId", is(task.getTaskId().intValue())))
```
Comment (5v3nn) [inline src/test/java/ch/uzh/ifi/hase/soprafs24/controller/TeamControllerTest.java:730]: check length

File: src/main/java/ch/uzh/ifi/hase/soprafs24/controller/TeamController.java [hunk 4]
```diff
@@ -144,6 +145,38 @@ public List<TaskGetDTO> getTasks(
...
+
+    // if tasks list is empty, return empty list
+    if (tasks.isEmpty()) {
+      return new ArrayList<>();
+    }
+
+    // convert internal representation of tasks back to API
+    return tasks.stream()
+        .map(DTOMapper.INSTANCE::convertEntityToTaskGetDTO)
+        .collect(Collectors.toList());
```
Comment (5v3nn) [inline src/main/java/ch/uzh/ifi/hase/soprafs24/controller/TeamController.java:177]: shouldn't empty lists be handled automatically? I.e., why check for `isEmtpy()` needed?

**Review by 5v3nn — CHANGES_REQUESTED** [review]

**Comment by 5v3nn** [comment]
@Alihan26 documentation is not done, please add in API specs what returns have been added/modified!
- [ ] Docu

File: src/main/java/ch/uzh/ifi/hase/soprafs24/controller/TeamController.java (hunk 4 above)
Comment (Alihan26) [reply src/main/java/ch/uzh/ifi/hase/soprafs24/controller/TeamController.java:177]: I don't know how to handle it automatically @5v3nn this was my way of implementation do I have to change it?

File: src/test/java/ch/uzh/ifi/hase/soprafs24/controller/TeamControllerTest.java (hunk 3 above)
Comment (Alihan26) [reply src/test/java/ch/uzh/ifi/hase/soprafs24/controller/TeamControllerTest.java:730]: Done

File: src/test/java/ch/uzh/ifi/hase/soprafs24/controller/TeamControllerTest.java (hunk 2 above)
Comment (Alihan26) [reply src/test/java/ch/uzh/ifi/hase/soprafs24/controller/TeamControllerTest.java:787]: yes, modified

**Comment by Alihan26** [comment]
> @Alihan26 documentation is not done, please add in API specs what returns have been added/modified!
> 
> * [ ]  Docu

I am on it

**Review by 5v3nn — APPROVED** [review]
very nice. sorry for the back and forth, but now we defenitely have something that will work consistently, no mater what the user does.

**Comment by 5v3nn** [comment]
very nice documentation on confluence. thank you

File: src/main/java/ch/uzh/ifi/hase/soprafs24/controller/TeamController.java (hunk 4 above)
Comment (5v3nn) [reply src/main/java/ch/uzh/ifi/hase/soprafs24/controller/TeamController.java:177]: ok if not possible

**Pull request merged**